packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -32,34 +148,52 @@ const parseDropEvent = ( event ) => {
 	}
 
 	return result;
-};
+}
 
-export default function useBlockDropZone( { element, rootClientId } ) {
-	const [ clientId, setClientId ] = useState( null );
+/**
+ * @typedef  {Object} WPBlockDropZoneConfig
+ * @property {Object} element      A React ref object pointing to the block list's DOM element.
+ * @property {string} rootClientId The root client id for the block list.
+ */
+
+/**
+ * A React hook that can be used to make a block list handle drag and drop.
+ *
+ * @param {WPBlockDropZoneConfig} dropZoneConfig configuration data for the drop zone.
+ *
+ * @return {number|undefined} The block index that's closest to the drag position.
+ */
+export default function useBlockDropZone( {
+	element,
+	rootClientId: targetRootClientId,
+} ) {
+	const [ targetBlockIndex, setTargetBlockIndex ] = useState( null );
 
 	function selector( select ) {
 		const {
 			getBlockIndex,
+			getBlockListSettings,
 			getClientIdsOfDescendants,
 			getSettings,
 			getTemplateLock,
 		} = select( 'core/block-editor' );
 		return {
 			getBlockIndex,
-			blockIndex: getBlockIndex( clientId, rootClientId ),
+			moverDirection: getBlockListSettings( targetRootClientId )
+				?.__experimentalMoverDirection,
```

Comment:
If this is merged, I think `__experimentalMoverDirection` could be renamed in a follow-up to something like `blockOrientation`, as it's now being used more widely than the movers.

Reply:
How about `blockListOrientation` or `blockFlowOrientation`? I agree `moverDirection` doesn't really fit anymore.